src/libcrun/container.c

```diff
@@ -1187,6 +1187,13 @@ container_init_setup (void *args, pid_t own_pid, char *notify_socket, int sync_s
   if (clearenv ())
     return crun_make_error (err, errno, "clearenv");
 
+  // set primary process to 1 explicitly if nothing is configured and LISTEN_FD is not set
```

Comment:
This seems different than what runc does: see https://github.com/opencontainers/runc/blob/8b59b768a945babf39e2468dc5d8afa54c511852/utils_linux.go#L279-L282

runc only sets if if LISTEN_FDS > 0

Reply:
It seems runc is always overriding when LISTEN_FDS > 0 but afaik for us `context->preserve_fds` will be only greater than `0` when `LISTEN_FDS` is set so behavior should be the same I think

Reply:
But if someone explicitly wants to set `LISTEN_PID` to something else i think we should preserve it unlike runc WDYT @giuseppe @vrothberg

Reply:
we can also drop `getenv("LISTEN_PID") == NULL` that should make everything equivalent to runc

Reply:
Maybe we should, @giuseppe WDYT?

Reply:
yes, Not sure if it is helpful, but IMO if LISTEN_PID is set to something else we should honor that.